### PR TITLE
[Demonology] Add support for `Fel Covenant` talent and 4 piece set bonus

### PR DESCRIPTION
**Pull request by darya-talanina**
Fel Covenant ([talent description on wowhead](https://www.wowhead.com/spell=387432/fel-covenant)): 

Screenshot 2022-12-29 at 22 56 52
Screenshot 2022-12-29 at 22 57 10

Four piece set bonus ([effect description on wowhead](https://www.wowhead.com/spell=394776/blazing-meteor)):

Screenshot 2022-12-29 at 22 56 22

Tested with [this log](https://www.warcraftlogs.com/reports/bPRACG3nj7gF1V8c):
* for Fel Covenant: Terros fight, Maelock character
* for set bonus: Raszageth, Maelock character

### CODE REVIEW

File: src/analysis/retail/warlock/demonology/modules/dragonflight/tier/VaultOfTheIncarnates4Set.tsx [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+import SPELLS from 'common/SPELLS';
+import Spell from 'common/SPELLS/Spell';
+import { TIERS } from 'game/TIERS';
+import Analyzer, { Options, SELECTED_PLAYER } from 'parser/core/Analyzer';
+import Events from 'parser/core/Events';
+import BoringSpellValueText from 'parser/ui/BoringSpellValueText';
+import Statistic from 'parser/ui/Statistic';
+import STATISTIC_CATEGORY from 'parser/ui/STATISTIC_CATEGORY';
+
+const BLAZING_METEOR: Spell = {
```
Comment (ToppleTheNun) [inline src/analysis/retail/warlock/demonology/modules/dragonflight/tier/VaultOfTheIncarnates4Set.tsx]: This should go in `src/common/SPELLS/warlock.ts`

File: src/analysis/retail/warlock/demonology/modules/talents/FelCovenant.tsx [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+class FelCovenant extends Analyzer {
+  static talent = TALENTS.FEL_COVENANT_TALENT;
+  static buffId = SPELLS.FEL_COVENANT_BUFF.id;
+
+  constructor(options: Options) {
+    super(options);
+
+    this.active = this.selectedCombatant.hasTalent(FelCovenant.talent);
+
+    if (!this.active) {
```
Comment (ToppleTheNun) [inline src/analysis/retail/warlock/demonology/modules/talents/FelCovenant.tsx]: This `if` check isn't necessary

**Review by ToppleTheNun — CHANGES_REQUESTED** [review]
Apart from two small things, this looks good!

File: src/analysis/retail/warlock/demonology/modules/dragonflight/tier/VaultOfTheIncarnates4Set.tsx (hunk 1 above)
Comment (darya-talanina) [reply src/analysis/retail/warlock/demonology/modules/dragonflight/tier/VaultOfTheIncarnates4Set.tsx]: Moved it.
I liked the idea of the spell being defined near the only place that uses it though.

File: src/analysis/retail/warlock/demonology/modules/talents/FelCovenant.tsx (hunk 2 above)
Comment (darya-talanina) [reply src/analysis/retail/warlock/demonology/modules/talents/FelCovenant.tsx]: Removed it. It does appear in a lot of other analyzers though.

Comment (ToppleTheNun) [reply src/analysis/retail/warlock/demonology/modules/talents/FelCovenant.tsx]: We've had mixed messaging on it in the past, so a lot of them have it hanging around

File: src/analysis/retail/warlock/demonology/modules/talents/FelCovenant.tsx [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+      },
+      style: ThresholdStyle.PERCENTAGE,
+    };
+  }
+
+  statistic() {
+    return (
+      <Statistic position={STATISTIC_ORDER.CORE()}>
+        <BoringSpellValueText spellId={FelCovenant.buffId}>
+          <>
```
Comment (ToppleTheNun) [inline src/analysis/retail/warlock/demonology/modules/talents/FelCovenant.tsx:42]: You don't need the fragment wrapper here, it's wrapped by `BoringSpellValueText`

**Review by ToppleTheNun — APPROVED** [review]

**Pull request merged**